frontend/public/components/graphs/prometheus-graph.tsx

```diff
@@ -19,23 +24,43 @@ export const getPrometheusExpressionBrowserURL = (urls, queries): string => {
   return `${base}/graph?${params.toString()}`;
 };
 
-const getQueryBrowserURL = (query: string): string => {
-  const params = new URLSearchParams();
-  params.set('query0', query);
-  return `/monitoring/query-browser?${params.toString()}`;
-};
+const mapStateToProps = (state: RootState) => ({
+  activePerspective: getActivePerspective(state),
+  canAccessMonitoring:
+    !!state[featureReducerName].get(FLAGS.CAN_GET_NS) && !!window.SERVER_FLAGS.prometheusBaseURL,
+  namespace: getActiveNamespace(state),
+});
 
-export const PrometheusGraphLink = ({
+export const PrometheusGraphLink_ = ({
+  activePerspective,
+  canAccessMonitoring,
   children,
+  namespace,
   query,
-}: React.PropsWithChildren<PrometheusGraphLinkProps>) =>
-  query ? (
-    <Link to={getQueryBrowserURL(query)} style={{ color: 'inherit', textDecoration: 'none' }}>
-      {children}
-    </Link>
-  ) : (
-    <>{children}</>
-  );
+}: React.PropsWithChildren<PrometheusGraphLinkProps>) => {
+  if (query && canAccessMonitoring) {
```

Comment:
We can render a link even if user cannot access full `/monitoring` section - we can switch to `dev` perspective `/metrics` which is accessible even for non-admin users. See how its implemented for Top Consumers view more link

See https://github.com/openshift/console/blob/master/frontend/packages/console-shared/src/components/dashboard/utilization-card/TopConsumerPopover.tsx#L145 and https://github.com/openshift/console/blob/master/frontend/packages/console-shared/src/components/dashboard/utilization-card/TopConsumerPopover.tsx#L150

Reply:
Thanks @rawagner. I reworked it to use the same approach as Top Consumers.